lib/proxy.js

```diff
@@ -174,7 +174,7 @@ function ReverseProxy(opts){
     if(target){
       proxy.ws(req, socket, head, {target: target});
     }else{
-      notFound(socket);
+      respondNotFound(req, socket);
```

Comment:
This isn't an HTTP response object, but does a socket have the same set of properties and methods as an HTTP response?